### PR TITLE
File upload example

### PR DESCRIPTION
**Pull request by DIRECTcut**
#2 "How to upload a file with TestCafe"
need to merge before [this ](https://github.com/DevExpress/testcafe/pull/5525)

### CODE REVIEW

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: examples/upload-files/README.MD [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Select Files To Upload
+
+**Test Code**: [index.js](index.js)
+
+**Tested Page**: [index.html](index.html)
+
+TestCafe cannot test native browser dialogs, including those during file selection. For this reason, TestCafe utilizes another approach to file selection. This example shows how to select files to populate a `type="file"` input.
+
+The tested page includes an `<input>` and a `<button>`. The input accepts multiple files. A `<button>` click [fetches](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) the files to the server in a 'POST' request.
```
[the remark this answers is not in the document]
Comment (miherlosev) [reply examples/upload-files/README.MD]: Fixed.

File: examples/upload-files/README.MD [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+**Test Code**: [index.js](index.js)
+
+**Tested Page**: [index.html](index.html)
+
+TestCafe cannot test native browser dialogs, including those during file selection. For this reason, TestCafe utilizes another approach to file selection. This example shows how to select files to populate a `type="file"` input.
+
+The sample page includes a `<form>` that contains:
+
+* `<input>` element that holds the selected files
+* the `submit` button that sends the files to the server.
```
Comment (miherlosev) [inline examples/upload-files/README.MD:12]: `Submit button` is a button with type `submit` (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button).

`input with type=submit that sends the files to the server`
here and below.

File: examples/upload-files/README.MD [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+During the test the `t.setFilesToUpload` method selects the files to send through the `<input>`. A `submit` button click sends the files. The server responds with a list of uploaded files, which is evaluated with the `t.expect.eql` Method.
```
Comment (miherlosev) [inline examples/upload-files/README.MD:14]: During the test the `t.setFilesToUpload` method set the files for the `<input type=file`. A `input type=submit` initiates the sending files to the server. The server responds with a list of uploaded files, which is checked with the `t.expect.eql` method.

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: examples/upload-files/README.MD (hunk 3 above)
Comment (DIRECTcut) [reply examples/upload-files/README.MD:14]: > During the test the `t.setFilesToUpload` method sets the files for the `<input type="file">`. A click on the `<input type="submit">` initiates the file upload to the server. The server responds with a list of uploaded files, which is checked with the `t.expect.eql` Method.

File: examples/upload-files/README.MD (hunk 2 above)
Comment (DIRECTcut) [reply examples/upload-files/README.MD:12]: edited

File: examples/upload-files/README.MD [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Select Files To Upload
+
+**Test Code**: [index.js](index.js)
+
+**Tested Page**: [index.html](index.html)
+
+TestCafe cannot test native browser dialogs, including those during file selection. For this reason, TestCafe utilizes another approach to file selection. This example shows how to select files to populate a `type="file"` input.
```
[the remark this answers is not in the document]
Comment (DIRECTcut) [reply examples/upload-files/README.MD:7]: > TestCafe cannot access the browser's `Choose File` dialog to emulate file selection within it. For this reason, TestCafe utilizes another approach to file selection. This example shows how to select files to populate a `type="file"` input.

File: examples/upload-files/index.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `http://localhost:3000/examples/upload-files/index.html`;
+
+test('Select files to upload', async t => {
```
[the remark this answers is not in the document]
Comment (DIRECTcut) [reply examples/upload-files/index.js:6]: edited

File: examples/upload-files/README.MD (hunk 3 above)
Comment (miherlosev) [reply examples/upload-files/README.MD]: Ok

**Review by miherlosev — APPROVED** [review]

File: examples/upload-files/README.MD [hunk 6]
```diff
@@ -0,0 +1,25 @@
+# Select Files To Upload
+
+**Test Code**: [index.js](index.js)
+
+**Tested Page**: [index.html](index.html)
+
+TestCafe cannot access the browser's `Choose File` dialog to emulate file selection. For this reason, TestCafe uses another way to select files. This example shows how to select files to populate a `type="file"` input.
```
Comment (VasilyStrelyaev) [inline examples/upload-files/README.MD:7]: > For this reason, TestCafe uses another way to select files.

After this sentence, I would expect some words about this other way (it's intriguing). However, we say nothing...

I think that the following phrasing creates less intrigue:

*TestCafe cannot access the browser's `Choose File` dialog to emulate file selection. This example shows how to add files to a `type="file"` input without browser dialogs.*

File: examples/upload-files/README.MD [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+The sample page includes a `<form>` that contains:
+
+* an `<input>` with `type="file"` that holds the selected files
+* an `<input>` with `type="submit"` that sends the files to the server
+
+During the test the `t.setFilesToUpload` method sets the files for the `<input type="file">`. A click on the `<input type="submit">` initiates the file upload to the server. The server responds with a list of uploaded files, which is checked with the `t.expect.eql` Method.
```
Comment (VasilyStrelyaev) [inline examples/upload-files/README.MD:14]: > A click on the `<input type="submit">` initiates the file upload to the server.

It's not really bad, but still, a little (quite a bit) weird to say that a *click does something* in technical docs. It's kind of figurative a bit.

*The test clicks the `<input type="submit">` to initiate the file upload.*

File: examples/upload-files/README.MD [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+## TestCafe Functions and Methods Used in This Example
```
Comment (VasilyStrelyaev) [inline examples/upload-files/README.MD:16]: Again, the header

**Review by VasilyStrelyaev — CHANGES_REQUESTED** [review]

File: examples/upload-files/README.MD (hunk 6 above)
Comment (DIRECTcut) [reply examples/upload-files/README.MD:7]: I agree. Edited

File: examples/upload-files/README.MD (hunk 7 above)
Comment (DIRECTcut) [reply examples/upload-files/README.MD:14]: Edited

File: examples/upload-files/README.MD (hunk 8 above)
Comment (DIRECTcut) [reply examples/upload-files/README.MD:16]: edited

**Review by VasilyStrelyaev — APPROVED** [review]

File: examples/upload-files/README.MD [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+**Tested Page**: [index.html](index.html)
+
+TestCafe cannot access the browser's `Choose File` dialog to emulate file selection. This example shows how to add files to a `type="file"` input without browser dialogs.
+
+The sample page includes a `<form>` that contains:
+
+* an `<input>` with `type="file"` that holds the selected files
+* an `<input>` with `type="submit"` that sends the files to the server
+
+During the test the `t.setFilesToUpload` method sets the files for the `<input type="file">`. The test clicks the `<input type="submit">` to initiates the file upload. The server responds with a list of uploaded files, which is checked with the `t.expect.eql` Method.
```
[the remark this answers is not in the document]
Comment (DIRECTcut) [reply examples/upload-files/README.MD:14]: During the test the `t.setFilesToUpload` method **specifies the file paths** for the `<input type="file">` element. The test clicks the `<input type="submit">` to **initiate** the file upload. Then the test receives a list of uploaded files from the server and checks it with the `t.expect.eql` Method.

**Pull request merged**